File: Practice/17/Python/ex17.py
maximum, Max, n = 0, -1, 1
black = red = 0
amount = [0] * 37
while n > 0:
    n = int(input())
    if n > 36:
        print("Число должно быть не больше 36", end = '')
        break;
    amount[n] += 1;
    if (n == 1 or n == 3 or n == 5 or n == 7 or n == 9 or n == 12 or n == 14 or n == 16 or n == 18 or n == 19 or n == 21 or n == 23 or n == 25 or n == 27 or n == 30 or n == 32 or n == 34 or n == 36):
        red += 1
    elif (n != 0):
        black += 1
    i = 0
    while i <= 36:
        if amount[i] > maximum:
            maximum = amount[i]
        i += 1
    i = 0
    while i <= 36:
        Max += 1
        if amount[i] == maximum:
            print(str(Max), end = ' ')
        i += 1
    Max = -1
    i = 0
    print('')
    while i <= 36:
        if n == i:
            print('', end = '')
        else:
            print(str(i), end = ' ')
        i += 1
    print('\n' + str(red) + ' ' + str(black) + '\n')

File: Practice/17/C++/ex17/ex17/ex17.cpp
#include <iostream>

int maximum = 0, Max = -1, n = 1, black = 0, red = 0, amount[37];

int main() {
    setlocale(LC_ALL, "ru");
    while (n > 0) {
        std::cin >> n;
        if (n > 36) {
            std::cout << "Число должно быть не больше 36";
            return 0;
        }
        amount[n] += 1;
        if (n == 1 || n == 3 || n == 5 || n == 7 || n == 9 || n == 12 || n == 14 || n == 16 || n == 18 || n == 19 || n == 21 || n == 23 || n == 25 || n == 27 || n == 30 || n == 32 || n == 34 || n == 36) {
            red++;
        } else if (n != 0) {
           black++; 
        } 
        for (int i = 0; i <= 36; i++)
        {
            if (amount[i] > maximum) {
                maximum = amount[i];
            }
        }
        for (int i = 0; i <= 36; i++)
        {
            Max++;
            if (amount[i] == maximum) {
                std::cout << Max << ' ';
            }
        }
        Max = -1;
        std::cout << '\n';
        for (int i = 0; i <= 36; i++) {
            if (n == i) {
                std::cout << "";
            }
            else {
                std::cout << i << ' ';
            }
        }
        std::cout << std::endl << red << ' ' << black << std::endl;
        std::cout << '\n';
    }
}